### PR TITLE
feat(server): add structured logging

### PR DESCRIPTION
**Pull request by filariow**

### CODE REVIEW

File: server/persistence/kube/kube.go [hunk 1]
```diff
@@ -42,7 +41,6 @@ func (c *Client) CreateUserWorkspace(ctx context.Context, user string, workspace
 	}()
 
 	workspace.SetNamespace(c.workspacesNamespace)
-	log.Printf("cli.Create %v ", workspace)
```
Comment (sadlerap) [inline server/persistence/kube/kube.go:45]: I would like to have some kind of logging here; would get some clarity from proxy on what it's thinking it's creating.

File: server/persistence/kube/kube.go [hunk 2]
```diff
@@ -58,7 +56,6 @@ func (c *Client) UpdateUserWorkspace(ctx context.Context, user string, workspace
 	}()
 
 	workspace.SetNamespace(c.workspacesNamespace)
-	log.Printf("cli.Update %v ", workspace)
```
Comment (sadlerap) [inline server/persistence/kube/kube.go:61]: Same here.

File: server/rest/server.go [hunk 3]
```diff
@@ -25,12 +27,13 @@ func New(
 ) *http.Server {
 	return &http.Server{
 		Addr:              addr,
-		Handler:           buildServerHandler(readHandle, listHandle, createHandle, updateHandle),
+		Handler:           buildServerHandler(logger, readHandle, listHandle, createHandle, updateHandle),
```
Comment (sadlerap) [inline server/rest/server.go:30]: It's at this point where I'm beginning to think we might want a builder pattern for our handler here.  I wouldn't expect it for this PR, but I might end up doing this in the future.  Thoughts?

**Review by sadlerap — COMMENTED** [review]
A thought: we already use `logr` over in `controller/`, but we're moving to `slog` here.  Long-term, I'd prefer us to use one logging framework and stick to it across everything.

Last I looked at the documentation, `logr` has support for `slog` as a backend (they export a struct that implements `Handler`), so maybe that's what we use long-term?

I'll do some more research into this on Monday; I want to see what kind of logging this produces (as compared to what `log` spits out).

File: server/persistence/kube/kube.go (hunk 2 above)
Comment (filariow) [reply server/persistence/kube/kube.go:61]: Added in 01e7d62c3d019785ff80f98b2bc24a25a46d210c

File: server/persistence/kube/kube.go (hunk 1 above)
Comment (filariow) [reply server/persistence/kube/kube.go:45]: Added in 01e7d62c3d019785ff80f98b2bc24a25a46d210c

**Review by sadlerap — APPROVED** [review]
Alright, after playing around with `slog` a bit, I'm not opposed to using it.  I rather like the API we get TBH.  LGTM.

One follow-up question, however: do you know if we have any restrictions on an output format for our logs?  I know in operators logs are usually in a JSON format, but I don't know if we have any requirement to do so here.

**Comment by filariow** [comment]
> One follow-up question, however: do you know if we have any restrictions on an output format for our logs? I know in operators logs are usually in a JSON format, but I don't know if we have any requirement to do so here.

As of now, I don't know which is the format we'll need to use. I'd guess JSON format.

**Pull request merged**